tests/e2e/nvidia-ctk_containerd_test.go

```diff
@@ -0,0 +1,806 @@
+			})
+
+			AfterEach(func(ctx context.Context) {
+				// Step 1: Restore containerd configuration from backup
+				_, _, err := nestedContainerRunner.Run(`
+			# Restore the original conf.d
+			if [ -d /tmp/containerd-conf.d.backup ]; then
+				rm -rf /etc/containerd/conf.d
+				cp -r /tmp/containerd-conf.d.backup /etc/containerd/conf.d
+			else
+				# If no backup exists, just clean up
+				rm -rf /etc/containerd/conf.d
+				mkdir -p /etc/containerd/conf.d
+			fi
+
+			# Restore the original config.toml
+			if [ -f /tmp/containerd-config.toml.backup ]; then
+				cp /tmp/containerd-config.toml.backup /etc/containerd/config.toml
+			else
+				# No backup means there was no original config file, so remove it
+				rm -f /etc/containerd/config.toml
+			fi
+		`)
+				Expect(err).ToNot(HaveOccurred(), "Failed to restore containerd configuration")
+
+				// Step 2: Restart containerd and wait for it to be ready
+				err = restartContainerdAndWait(nestedContainerRunner)
+				Expect(err).ToNot(HaveOccurred(), "Failed to restart containerd after configuration restore")
+			})
+
+			When("configuring containerd", func() {
+				It("should add NVIDIA runtime using drop-in config and validate merged configuration", func(ctx context.Context) {
+					// Configure containerd using nvidia-ctk
+					_, _, err := nestedContainerRunner.Run(`nvidia-ctk runtime configure --runtime=containerd --set-as-default --cdi.enabled`)
+					Expect(err).ToNot(HaveOccurred(), "Failed to configure containerd")
+
+					// Verify drop-in config was created
+					dropInConfigContents, _, err := nestedContainerRunner.Run("cat /etc/containerd/conf.d/99-nvidia.toml")
+					Expect(err).ToNot(HaveOccurred())
+					Expect(dropInConfigContents).ToNot(BeEmpty())
+
+					// Verify main config has imports directive
+					mainConfigContents, _, err := nestedContainerRunner.Run("cat /etc/containerd/config.toml")
+					Expect(err).ToNot(HaveOccurred())
+					Expect(mainConfigContents).To(ContainSubstring("imports"))
+					Expect(mainConfigContents).To(ContainSubstring("/etc/containerd/conf.d"))
+
+					// Restart containerd
+					err = restartContainerdAndWait(nestedContainerRunner)
+					Expect(err).ToNot(HaveOccurred(), "Failed to restart containerd")
+
+					// Verify merged config structure
+					output, _, err := nestedContainerRunner.Run(`containerd config dump`)
+					Expect(err).ToNot(HaveOccurred())
+
+					// Parse the TOML output
+					config, err := parseContainerdConfig(output)
+					Expect(err).ToNot(HaveOccurred(), "Failed to parse containerd config")
+
+					// Verify config version
+					Expect(config.Get("version")).To(Equal(env.configVersion))
+
+					// Verify imports are present in the merged config
+					imports := config.Get("imports")
+					Expect(imports).ToNot(BeNil())
+					importsList, ok := imports.([]interface{})
+					Expect(ok).To(BeTrue())
+					Expect(len(importsList)).To(BeNumerically(">=", 1))
```

Comment:
Checking for the presence of imports is not sufficient. We need to ensure that we ALSO check that the desired folder (or file) is imported. For example, see the matcher here: https://github.com/ArangoGutierrez/nvidia-container-toolkit/pull/1/files#diff-056e23fcacb20f063e793331c10a8fedec7eedb12c88a3c7e857ad995ace49daR234-R249

(I'm not saying that we need to use a matcher, but that we should check the value properly).